lib/beacon/loader/data_source_module_loader.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Beacon.Loader.DataSourceModuleLoader do
+  @moduledoc false
+  alias Beacon.Loader
+
+  require Logger
+
+  def load_data_source(data, site) do
+    data_source_module = Loader.data_source_module_for_site(site)
+    live_data_functions = Enum.map(data, &live_data_fn/1)
+    # TODO default data
+
+    ast =
+      quote do
+        defmodule unquote(data_source_module) do
+          @behaviour Beacon.DataSource.Behaviour
+          require Logger
+
+          @impl Beacon.DataSource.Behaviour
+          def live_data(path, params) do
+            live_data(path, params, %{})
+          end
+
+          unquote_splicing(live_data_functions)
+
+          def live_data(path, params, data) do
+            Logger.warning("Unhandled Beacon Live Data request for site #{unquote(site)} with path #{inspect(path)} and params #{inspect(params)}")
+            data
+          end
+        end
+      end
+
+    # For debugging - will print module content to the terminal
```

Comment:
Since we debug AST very often, this block of code seems a good candidate to live in `Beacon.Utils` which is a private module used internally